keel-actuator/src/main/kotlin/com/netflix/spinnaker/keel/actuation/ResourceActuator.kt

```diff
@@ -65,45 +62,37 @@ class ResourceActuator(
       when {
         current == null -> {
           log.warn("Resource {} is missing", name)
-          publisher.publishEvent(ResourceChecked(resource, Missing))
-
-          resourceRepository.appendHistory(ResourceMissing(resource, clock))
+          publisher.publishEvent(ResourceMissing(resource, clock))
 
           plugin.create(resource, diff)
             .also { tasks ->
-              resourceRepository.appendHistory(ResourceActuationLaunched(resource, plugin.name, tasks, clock))
+              publisher.publishEvent(ResourceActuationLaunched(resource, plugin.name, tasks, clock))
             }
         }
         diff.hasChanges() -> {
           log.warn("Resource {} is invalid", name)
           log.info("Resource {} delta: {}", name, diff.toDebug())
-          publisher.publishEvent(ResourceChecked(resource, Diff))
-
-          resourceRepository.appendHistory(ResourceDeltaDetected(resource, diff.toDeltaJson(), clock))
+          publisher.publishEvent(ResourceDeltaDetected(resource, diff.toDeltaJson(), clock))
 
           plugin.update(resource, diff)
             .also { tasks ->
-              resourceRepository.appendHistory(ResourceActuationLaunched(resource, plugin.name, tasks, clock))
+              publisher.publishEvent(ResourceActuationLaunched(resource, plugin.name, tasks, clock))
             }
         }
         else -> {
           log.info("Resource {} is valid", name)
+          // TODO: not sure this logic belongs here
           val lastEvent = resourceRepository.eventHistory(resource.uid).first()
           if (lastEvent is ResourceDeltaDetected || lastEvent is ResourceActuationLaunched) {
-            val event = ResourceDeltaResolved(
-              resource = resource,
-              current = current,
-              clock = clock
-            )
-            resourceRepository.appendHistory(event)
-            publisher.publishEvent(event) // TODO: refactor pending to unify how all this stuff with events works
```

Comment:
This was the case that really triggered me wanting to do this refactor. It seems crazy for this class to be responsible for appending to the resource history _and_ raising an event for the benefit of another listener. Instead the resource history can just be managed by a dedicated listener responding to the same events.